src/test/java/com/ibm/cloud/sdk/core/test/security/ConfigBasedAuthenticatorFactoryTest.java

```diff
@@ -310,12 +333,48 @@ public void testFileCredentialsService17() {
   }
 
   @Test
-  public void testFileCredentialsService5() {
```

Comment:
It's not totally obvious, but I moved "testFileCredentialsService5" up near "service4" and added three new test methods here.